pkg/common/versions/version_selector.go

```diff
@@ -33,6 +36,11 @@ func GetVersionForInstall(versionList *spi.VersionList) (*semver.Version, string
 
 	v, selector, err := selectedVersionSelector.SelectVersion(versionList)
 
+	channel := viper.GetString(config.Cluster.Channel)
+	if channel != "stable" && !strings.Contains(v.Original(), channel) {
+		v = semver.MustParse(fmt.Sprintf("%s-%s", v.Original(), channel))
```

Comment:
Doesn't that allow arbitrary input as long as it's semver formatted?

Reply:
Arbitrary input for supplying the channel? Yes. But that goes for almost all of our config options.

Reply:
Was just curious if I understood correctly. Sorry if the comment was misleading